core/providers/azure/images.go

```diff
@@ -0,0 +1,302 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/bytedance/sonic"
+	"github.com/google/uuid"
+	"github.com/maximhq/bifrost/core/providers/openai"
+	providerUtils "github.com/maximhq/bifrost/core/providers/utils"
+	"github.com/maximhq/bifrost/core/schemas"
+)
+
+// AzureImageRequest is the struct for Image Generation requests by Azure.
+type AzureImageRequest struct {
+	Model          string  `json:"model"`
+	Prompt         string  `json:"prompt"`
+	N              *int    `json:"n,omitempty"`
+	Size           *string `json:"size,omitempty"`
+	Quality        *string `json:"quality,omitempty"`
+	Style          *string `json:"style,omitempty"`
+	ResponseFormat *string `json:"response_format,omitempty"`
+	User           *string `json:"user,omitempty"`
+}
+
+// AzureImageResponse is the struct for Image Generation responses by Azure.
+type AzureImageResponse struct {
+	Created int64 `json:"created"`
+	Data    []struct {
+		URL           string `json:"url,omitempty"`
+		B64JSON       string `json:"b64_json,omitempty"`
+		RevisedPrompt string `json:"revised_prompt,omitempty"`
+	} `json:"data"`
+	Usage *AzureImageGenerationUsage `json:"usage"`
+}
+
+type AzureImageGenerationUsage struct {
+	TotalTokens  int `json:"total_tokens,omitempty"`
+	InputTokens  int `json:"input_tokens,omitempty"`
+	OutputTokens int `json:"output_tokens,omitempty"`
+
+	InputTokensDetails *struct {
+		TextTokens  int `json:"text_tokens,omitempty"`
+		ImageTokens int `json:"image_tokens,omitempty"`
+	} `json:"input_tokens_details,omitempty"`
+}
+
+// ImageGeneration performs an Image Generation request to Azure's API.
+// It formats the request, sends it to Azure, and processes the response.
+// Returns a BifrostResponse containing the bifrost response or an error if the request fails.
+func (provider *AzureProvider) ImageGeneration(ctx context.Context, key schemas.Key,
```

Comment:
we follow a convention of having all provider interface methods in `provider/provider.go` file and only request and response convertors in operation.go file (which here is images.go) - can you pls follow this for all providers

Reply:
and all structs in `provider/types.go`